File: General/bottles-of-beer.js
(function () {
  'use strict'
  function capitalize (word) {
    return word.charAt(0).toUpperCase() + word.slice(1)
  }
  function beer (bottles) {
    for (var bottle = bottles; bottle > -1; bottle--) {
      var word = (bottle === 1) ? 'bottle' : 'bottles'
      console.log(`${bottle} ${word} of beer on the wall, ${bottle} ${word} of beer on the wall.`)
      console.log(`Take one down, pass it around, ${bottle} ${word} of beer on the wall.`)
    }
  }
  beer(99)
}())

File: General/bottles-of-beer.rb
def beer (bottles)
  (bottles).downto(0) do |bottle|
    word = bottle == 1 ? 'bottle' : 'bottles'
    bottle = bottle == 0 ? 'no more' : bottle
    puts "#{(bottle.is_a? String) ? bottle.capitalize : bottle} #{word} of beer on the wall, #{bottle} #{word} of beer."
    puts "Take one down, pass it around, #{(bottle.is_a? String) ? bottle : (bottle - 1)} #{word} of beer on the wall."
  end
end

beer(99)
